lib/chef/provider/package/openbsd.rb

```diff
@@ -57,7 +57,7 @@ def define_resource_requirements
           end
           requirements.assert(:all_actions) do |a|
             a.assertion do
-              if new_resource.package_name =~ /^(.+?)--(.+)/
+              if new_resource.package_name =~ /^(.+?)--(.?+)/
```

Comment:
i'm slightly puzzled by the regular expression here since it changes it so that "foo--bar" will return "b" instead of "bar" because the match is now non-greedy, which doesn't match what i'd expect for "empty flavor" since I would have expected just ".*" or maybe "(?:--(.*))" or something like that...  definitely seems to need either some better comment documentation or ideally some tests.